lib/transaction/sidechain.js

```diff
@@ -266,65 +271,65 @@ Sidechain.fromBufferReader = function(br) {
     mainchainBackwardTransferScFeeBN = tmp[1];
     mbtrRequestDataLength = br.readUInt8();
 
-    var vsc = new Vsc_ccout(withdrawal_epoch_length, satoshis, satoshisBN, address, customData, constantData, certVk,
-        wCeasedVk, vFieldElementCertificateFieldConfig, vBitVectorCertificateFieldConfig, forwardTransferScFee, forwardTransferScFeeBN,
-        mainchainBackwardTransferScFee, mainchainBackwardTransferScFeeBN, mbtrRequestDataLength);
+    const vsc = new Vsc_ccout(withdrawal_epoch_length, satoshis, satoshisBN, address, customData, constantData, certVk,
+      wCeasedVk, vFieldElementCertificateFieldConfig, vBitVectorCertificateFieldConfig, forwardTransferScFee,
+      forwardTransferScFeeBN, mainchainBackwardTransferScFee, mainchainBackwardTransferScFeeBN, mbtrRequestDataLength);
     sc.vsc_ccout.push(vsc);
   }
 
   // Read Vft_ccout
-  var n_vft = br.readVarintNum();
+  const n_vft = br.readVarintNum();
   for (i = 0; i < n_vft; i++) {
-    var tmp = this._checkAmount(br.readUInt64LEBN());
+    const tmp = this._checkAmount(br.readUInt64LEBN());
     satoshis = tmp[0];
     satoshisBN = tmp[1];
     address = br.read(32);
-    var scid = br.read(32);
-    var vft = new Vft_ccout(satoshis, satoshisBN, address, scid);
+    scid = br.read(32);
+    mcReturnAddress = br.read(20);
+    const vft = new Vft_ccout(satoshis, satoshisBN, address, scid, mcReturnAddress);
     sc.vft_ccout.push(vft);
   }
 
-  var n_vmbtr_out = br.readVarintNum();
-  var scid, pubKeyHash, scFee, scFeeBN;
-  for (i=0; i<n_vmbtr_out; i++) {
+  const n_vmbtr_out = br.readVarintNum();
+  for (i = 0; i < n_vmbtr_out; i++) {
     scid = br.read(32);
-    var n_vScRequestData = br.readVarintNum();
-    var vScRequestData = []
+    const n_vScRequestData = br.readVarintNum();
+    const vScRequestData = [];
     for (j = 0; j < n_vScRequestData; j++) {
-      var len_vScRequestData = br.readVarintNum();
+      const len_vScRequestData = br.readVarintNum();
       vScRequestData.push(br.read(len_vScRequestData));
     }
-    pubKeyHash = br.read(20);
-    tmp = this._checkAmount(br.readUInt64LEBN());
+    mcDestinationAddress = br.read(20);
```

Comment:
In zendoo this field is named "pubkeyhash", i suggest to keep this name

Reply:
Should we also change it in the `class Vmbtr_out` and other related places?

Reply:
notes on code